### PR TITLE
US018 Edit Job Family

### PR DESCRIPTION
**Pull request by mactrz**
As an Admin
I want to edit a job family
So that I can keep the software information up to date

Acceptance Criteria

Able to edit job family title

### CODE REVIEW

**Review by michaltangri — APPROVED** [review]

File: routes/jobFamilyRoutes.js [hunk 1]
```diff
@@ -1,9 +1,34 @@
...
+router.get("/:capabilityName/edit/:jobFamilyName", async (req, res) => {
+    const capabilityName = req.params.capabilityName;
+    const jobFamilyName = req.params.jobFamilyName;
+    const jobFamilies = await getJobFamiliesByCapabilityName(capabilityName);
+    if(jobFamilies.length > 0) {
+        if(jobFamilies.findIndex(x => x.name === jobFamilyName) !== -1) {
+            res.render('editJobFamily', {jobFamilyName: jobFamilyName, capabilityName});
+        }
+        else {
+            res.render('notFound');
```
Comment (TheoHodges) [inline routes/jobFamilyRoutes.js:24]: There is a lot of space here taken up by braces. Consider changing the if/else statements to not use braces as they only have one line inside them.

File: selenium/jobFamily.test.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    const jobFamilyInput = await driver.findElement(webdriver.By.name('newJobFamilyName'));
+    const submitButton = await driver.findElement(webdriver.By.id('submit-button'));
+    await jobFamilyInput.sendKeys('Architecture');
+    await submitButton.click();
+    const button = await driver.wait(webdriver.until.elementLocated(webdriver.By.css('body > div > div > div > form > button'), 5000));
+    let buttonText = await button.getText();
+    expect(buttonText).toBe('Submit');
+ });
+}, 30000); 
+});
```
Comment (TheoHodges) [inline selenium/jobFamily.test.js:47]: Indentation looks a bit dodgy around here. Also maybe remove the 30000 magic number and replacing it with a constant and add a newline at the end of the file.

File: services/jobFamilyService.js [hunk 3]
```diff
@@ -11,3 +11,15 @@ exports.getJobFamiliesByCapabilityName = async (capabilityName) => {
     })
     return results
 }
+
+
```
Comment (TheoHodges) [inline services/jobFamilyService.js:15]: Whitespace police

File: tests/jobFamily.test.js [hunk 4]
```diff
@@ -53,4 +54,28 @@ describe("Test the job family service", () => {
...
+    test("updateJobFamilyName should return expected response", async () => {
+        mockAxios.patch.mockImplementation(() => {
+            return Promise.resolve({data: "value"});
+        });
+
+        let results = await updateJobFamilyName("n1", "n2");
+
+        expect(results).toBe("value");
+
+        expect(mockAxios.patch).toHaveBeenCalledWith(`${process.env.API_HOST}/job-family/n1/n2`);
```
Comment (TheoHodges) [inline tests/jobFamily.test.js:66]: Whitespace police

File: views/jobFamily.njk [hunk 5]
```diff
@@ -26,6 +26,14 @@
...
+                    <td>
+                        <a href="/job-family/{{jobFamily.capability.name}}/edit/{{jobFamily.name}}">
+                            <svg xmlns="http://www.w3.org/2000/svg" width="16" height="16" fill="currentColor" class="bi bi-pencil-square" viewBox="0 0 16 16">
+                                <path d="M15.502 1.94a.5.5 0 0 1 0 .706L14.459 3.69l-2-2L13.502.646a.5.5 0 0 1 .707 0l1.293 1.293zm-1.75 2.456-2-2L4.939 9.21a.5.5 0 0 0-.121.196l-.805 2.414a.25.25 0 0 0 .316.316l2.414-.805a.5.5 0 0 0 .196-.12l6.813-6.814z"/>
+                                <path fill-rule="evenodd" d="M1 13.5A1.5 1.5 0 0 0 2.5 15h11a1.5 1.5 0 0 0 1.5-1.5v-6a.5.5 0 0 0-1 0v6a.5.5 0 0 1-.5.5h-11a.5.5 0 0 1-.5-.5v-11a.5.5 0 0 1 .5-.5H9a.5.5 0 0 0 0-1H2.5A1.5 1.5 0 0 0 1 2.5v11z"/>
+                            </svg>
+                        <a/>
+                    </td>
                 </tr>
                 {%endfor%}
```
Comment (TheoHodges) [inline views/jobFamily.njk:38]: Dodgy indentation with the `{%endfor%}` and `</table>`

**Review by TheoHodges — CHANGES_REQUESTED** [review]

File: selenium/jobFamily.test.js (hunk 2 above)
Comment (sebastianczajkowski) [reply selenium/jobFamily.test.js]: I did that in the conflict resolve

File: routes/jobFamilyRoutes.js (hunk 1 above)
Comment (mactrz) [reply routes/jobFamilyRoutes.js:29]: I thought that leaving the braces can improve visibility, but removed as requested

File: selenium/jobFamily.test.js (hunk 2 above)
Comment (mactrz) [reply selenium/jobFamily.test.js]: Should be fixed

File: tests/jobFamily.test.js (hunk 4 above)
Comment (mactrz) [reply tests/jobFamily.test.js:75]: Gone

File: views/jobFamily.njk (hunk 5 above)
Comment (mactrz) [reply views/jobFamily.njk]: Not sure what happened here 😆 but should be fixed

**Review by TheoHodges — APPROVED** [review]

**Pull request merged**